### ISSUE TITLE
Problem z połączeniem 

### ISSUE THREAD

**Issue by dontbesoshy** [allegro/allegro-api#9805]
Mam problem z połączeniem się z danymi produkcyjnymi. 

Na środowisku sandbox bez problemu tworzę integrację. Na prod nie mogę. Mam błąd:

`Error with Allegro`s external API: HTTP request returned status code 401: {"error":"Unauthorized","error_description":"Unauthorized"}`

Sprawdzałem client id oraz secret - są dobre. Gdzie leży problem?

**Comment by PrzemyslawLukanowski** [comment]
Czy na pewno korzystasz z client Id i client secret przeznaczonych dla środowiska testowego, zarejestrowanych na stronie https://apps.developer.allegro.pl.allegrosandbox.pl/? Czy adresy do autoryzacji to na pewno https://allegro.pl.allegrosandbox.pl/auth/oauth/? Jeśli tak, a problem i tak występuje, podaj proszę swoje clientId.

**Comment by dontbesoshy** [comment]
sandbox działa poprawnie, ale gdy zmienie client id i secret z konta już developerskiego na allegro to jest powyższy komunikat
client id: 555bc67f7c09460e94508a75dad7dd45

**Comment by PrzemyslawLukanowski** [comment]
Tak, przepraszam, odwróciłem środowiska. W logach nie widzimy żadnej informacji wskazującej na przyczynę błędu, dlatego prześlij nam pełen request i response przez [formularz kontaktowy.](https://allegro.pl/pomoc/kontakt) Dopisz, że dane dotyczą wątku #9805 i należy je przekazać zespołowi API. Możesz jeszcze upewnić się, że dane w Authorization: Basic zakodowane są prawidłowo. Czasem zdarza się omyłkowy nadmiarowy znak np. na końcu clientID.

**Comment by dontbesoshy** [comment]
w sandbox używam adresów:

https://api.allegro.pl.allegrosandbox.pl
https://allegro.pl.allegrosandbox.pl
https://allegro.pl.allegrosandbox.pl/auth/oauth/token
oraz redirect

na produkcji używam

https://api.allegro.pl
https://allegro.pl
https://allegro.pl/auth/oauth/token
oraz redirect

**Comment by PrzemyslawLukanowski** [comment]
Tak, to prawidłowe adresy. Tak jak wspomniałem, możesz jeszcze upewnić się, że dane Authorization: Basic zakodowane są prawidłowo. Jeśli nie stwierdzisz błędów, prześlij nam pełne logi za pomocą formularza kontaktowego.

**Comment by PrzemyslawLukanowski** [comment]
O 11:31 widzę w logach udaną próbę wygenerowania tokenów. Czy rzeczywiście problem był w kodowaniu znaków?

**Comment by dontbesoshy** [comment]
tak, stała reprezentująca url do auth/oauth/token była niewłaściwa
